### PR TITLE
Add kotlin code for divide and conquer

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.
屏幕截图 2024-02-27 170850
屏幕截图 2024-02-27 170906
屏幕截图 2024-02-27 170916

### CODE REVIEW

File: codes/kotlin/chapter_computational_complexity/iteration.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/**
+ * File: iteration.kt
```
Comment (krahets) [inline codes/kotlin/chapter_computational_complexity/iteration.kt:2]: This PR contains merged changes. Please remove them to make the changes clean.

**Comment by krahets** [comment]
I mean revert the changes, not delete the files. Now these deleted files will disappear in the remote repo if I merge this PR.

**Comment by curtishd** [comment]
oh, Sorry for my stupidity

**Comment by krahets** [comment]
> oh, Sorry for my stupidity

No problem. Let's create a new PR. It is recommended to merge the main branch to your local branch before submitting a new PR.

**Pull request closed without merging**